dspace-server-webapp/src/test/java/org/dspace/app/rest/CollectionHarvestSettingsControllerIT.java

```diff
@@ -0,0 +1,288 @@
+     * @return A JSONObject containing the given harvest settings
+     */
+    public JSONObject createHarvestSettingsJson(String harvestType,
+                                                String oaiSource, String oaiSetId, String metadataConfigId) {
+        JSONObject json = new JSONObject();
+        json.put("harvest_type", harvestType);
+        json.put("oai_source", oaiSource);
+        json.put("oai_set_id", oaiSetId);
+        json.put("metadata_config_id", metadataConfigId);
+        return json;
+    }
+
+    @Test
+    public void GetCollectionHarvestSettings() throws Exception {
+        String token = getAuthToken(admin.getEmail(), password);
+
+        List<Map<String,String>> configs = OAIHarvester.getAvailableMetadataFormats();
+
+        // Add harvest settings to collection
+        JSONObject json = createHarvestSettingsJson("METADATA_ONLY", "https://dspace.org/oai/request", "col_1721.1_114174", "dc");
+
+        getClient(token).perform(
+            put("/api/core/collections/" + collection.getID() + "/harvester")
+                .contentType("application/json")
+                .content(json.toString()))
+            .andExpect(status().isOk());
+
+        //Retrieve harvest settings
+        getClient(token).perform(
+            get("/api/core/collections/" + collection.getID() + "/harvester"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.harvest_type", is("METADATA_ONLY")))
+                .andExpect(jsonPath("$.oai_source", is("https://dspace.org/oai/request")))
+                .andExpect(jsonPath("$.oai_set_id", is("col_1721.1_114174")))
+                .andExpect(jsonPath("$.harvest_message", is(nullValue())))
+                .andExpect(jsonPath("$.metadata_config_id", is("dc")))
+                .andExpect(jsonPath("$.harvest_status", is("READY")))
+                .andExpect(jsonPath("$.harvest_start_time", is(nullValue())))
+                .andExpect(jsonPath("$.last_harvested", is(nullValue())))
+                .andExpect(jsonPath("$._links.self.href",
+                    endsWith("api/core/collections/" + collection.getID() + "/harvester")))
+                .andExpect(jsonPath("$._embedded.metadata_configs",  Matchers.allOf(
+                    MetadataConfigsMatcher.matchMetadataConfigs(configs)
+                )))
+                .andExpect(jsonPath("$._embedded.metadata_configs._links.self.href",
+                    endsWith("/api/config/harvestermetadata")));
+    }
+
+    @Test
+    public void GetCollectionHarvestSettingsIfNotAdmin() throws Exception {
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        getClient(token).perform(
+            put("/api/core/collections/" + collection.getID() + "/harvester")
+                .contentType("application/json"))
+            .andExpect(status().isForbidden());
+    }
+
+    @Test
+    public void GetAndPutCollectionHarvestSettingsIfUserHasWriteRights() throws Exception {
```

Comment:
We should also verify in a test that a user *without* WRITE rights cannot interact with the settings.